### DISCUSSION TITLE
Can I define a DG (P1) space on triangle mesh with dofs defined on middle points of the edges?

### DISCUSSION THREAD

**Discussion by lrtfm** [firedrakeproject/firedrake#2383] [Firedrake support] [answered]
Now, I know how to define a DG space 
```
mesh = RectangleMesh(10, 10, 1, 1)
V = FunctionSpace(mesh, 'DG', 1)
```
The DOFs of `V` are defined on vertex. 
Can I define another space with the DOFs of `V` defined on the middle points of the edges?

**Comment by colinjcotter** [comment]
You can get this with


```
FunctionSpace(mesh, BrokenElement(FiniteElement("CR", triangle, 1)))
```

**Reply by lrtfm** [reply]
@colinjcotter Thanks for your quick reply.
I tried it and found the right syntax should be
```
V = FunctionSpace(mesh, BrokenElement(FiniteElement('CR', triangle, 1)))
```

**Comment by colinjcotter** [comment, marked as answer]
Yes, I forgot FiniteElement, sorry!

On 12 Mar 2022, at 16:03, Yang Zongze ***@***.***> wrote:

﻿

@colinjcotter<https://github.com/colinjcotter> Thanks for your quick reply.
I tried it and found the right syntax should be

V = FunctionSpace(mesh, BrokenElement(FiniteElement('CR', triangle, 1)))


—
Reply to this email directly, view it on GitHub<https://github.com/firedrakeproject/firedrake/discussions/2383#discussioncomment-2347169>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABOSV4UYLVHJEYE34UPSWTDU7S55XANCNFSM5QR7ZMXA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by lrtfm** [reply]
I think I can mark this as the answer. Thanks!
